lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -315,6 +380,150 @@ struct MMAV3UseRegOperand
   }
 };
 
+// MMAV3's analog of HoistLayoutConversion, for operand A only; will make WarpGroupDot
+// accept operand A in registers instead of shmem.
+//
+//  local_alloc(elementwise(x)) ->
```

Comment:
I assume this describes the rewrite that is being performed? I think it could be a bit more accurate. I don't see the convert or the #dot_operand before the conversion, and it's not obvious what happened to the local_alloc.